### PR TITLE
improve: share the updater HTTP client stack

### LINKED ISSUE #3138: Drop the second HTTP client stack tauri-plugin-updater brings
## Outcome

The desktop compiles one reqwest/hyper stack.

## Why

`cargo tree -i reqwest@0.13` shows only `tauri-plugin-updater 2.10.1` while the workspace pins `reqwest 0.12.28`, so the desktop builds two client stacks (`hyper-rustls`, `rustls-platform-verifier`, `tower-http 0.6` and `0.7`, `infer 0.19` and `0.22`, `webpki-roots` twice). The plugin is a 290-package subtree used at two call sites.

## Options

1. Move the workspace to reqwest 0.13 (already in `Cargo.lock`), migrating call sites in server, cli, router and desktop.
2. Replace the plugin with a small updater on `minisign-verify` (already a dependency) and the workspace `reqwest`.

Either needs a full desktop compile; do it on a developer machine.

### PR DESCRIPTION
**Pull request by naingthet**
The desktop builds reqwest 0.12.28 for workspace clients and 0.13.4 for the Tauri updater. Pin the workspace to the already locked 0.13.4 version so the desktop resolves one reqwest/hyper client stack.

Enable query and form encoding explicitly, retain JSON, multipart, and streaming support, and keep the Tauri updater’s verification and installation code. HTTPS certificate verification now uses the platform trust store through rustls. Request call sites need no changes.

Add a lockfile regression test for duplicate reqwest, hyper, and hyper-rustls versions, and regenerate the third-party notices. The regression test fails against the original lockfile.

Validation:
- Full macOS desktop build: passed.
- Desktop and CLI `cargo check --all-targets`: passed.
- Provider replay contracts: 5 passed.
- Gateway token exchange, refresh, and refusal tests: 28 passed.
- HTTP dependency tests: 3 passed.
- Formatting, diff, and generated notices checks: passed.

Closes #3138.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -107,10 +107,11 @@ tokio = { version = "=1.53.1", features = [
     "time",
 ] }
 tokio-util = { version = "=0.7.19", features = ["io-util"] }
-# rustls-tls-webpki-roots (never native-tls) matches the reqwest stack; the
-# loopback chat socket itself is plain ws://.
+# Use rustls for WebSockets; the loopback chat socket itself is plain ws://.
 tokio-tungstenite = { version = "=0.30.0", default-features = false, features = ["connect", "handshake", "rustls-tls-webpki-roots"] }
-reqwest = { version = "=0.12.28", default-features = false, features = ["json", "multipart", "stream", "rustls-tls"] }
+# Share reqwest with tauri-plugin-updater. Rustls uses the platform trust store;
```
[the remark this answers is not in the document]
Comment (naingthet) [reply Cargo.toml]: This finding does not apply to the locked reqwest 0.13.4. Its `rustls` feature explicitly enables `dep:rustls-platform-verifier` alongside `__rustls-aws-lc-rs` and `__rustls` (Cargo.toml lines 168–172). The default client builder creates `rustls_platform_verifier::Verifier::new(provider.clone())` when no custom roots are supplied (src/async_impl/client.rs lines 748–758).

I also ran a temporary example through the workspace tidebreak-router dependency, without the updater and with proxies disabled. A request to https://example.com returned 200 OK. A request to https://self-signed.badssl.com failed with InvalidCertificate and the platform error “certificate is not trusted: -67843”. The workspace client accepts a valid public certificate and rejects a self-signed certificate. No TLS feature change is needed.
